src/raft_server.c

```diff
@@ -4982,35 +5011,32 @@ raft_server_state_machine_apply(struct raft_instance *ri)
     sink_bi = buffer_set_allocate_item(&ri->ri_buf_set[RAFT_BUF_SET_LARGE]);
     NIOVA_ASSERT(sink_bi);
 
-    /* Signify that the entry will be applied.  Prepare the last-applied values
-     * prior to entering raft_server_sm_apply_opt().
-     */
-    raft_server_last_applied_increment(ri, &reh);
 
     // Read the raft entry
     if (!reh.reh_leader_change_marker && reh.reh_data_size)
     {
-        int rc = raft_server_entry_read(ri, apply_idx,
+        int rc = raft_server_entry_read(ri, to_apply_idx,
                                         (char *)sink_bi->bi_iov.iov_base,
                                         reh.reh_data_size, NULL);
         DBG_RAFT_INSTANCE_FATAL_IF((rc), ri, "raft_server_entry_read(): %s",
                                    strerror(-rc));
     }
 
-    /*
-     * Use single entry of write suppliement for each rncr objects to
-     * make sure all coalesced write entries are part of same write supplement
-     * structure.
-     */
-    struct raft_net_sm_write_supplements coalesced_ws = {0, NULL};
     bool failed = false;
 
     uint32_t offset = 0;
     char *sink_buf = (char *)sink_bi->bi_iov.iov_base;
     char *reply_buf;
     struct raft_net_client_request_handle *rncr_ptr;
 
-    for (uint32_t i = 0;
+    //Increment the last applied idx in the soft state
+    //only if its a new apply
+    if (raft_server_new_apply_condition(ri))
+        raft_server_last_applied_increment(ri, &reh);
```

Comment:
I believe I mentioned this before but I think it's improper to update the `ri_last_applied_cumulative_crc` until the commit data has been applied.  Unfortunately, there's no good approach here that doesn't involve adding an array of `reh_crc[]` to `struct raft_entry_header` unless we resign ourselves to only updating `ri_last_applied_cumulative_crc` once the last coalesce entry has been applied (which I think is fine as well though I'm leaning towards simply tracking the crc per coalesce entry).